src/compas_fab/artists/base.py

```diff
@@ -35,31 +35,31 @@ class BaseRobotArtist(object):
     def __init__(self, robot):
         super(BaseRobotArtist, self).__init__()
         self.robot = robot
-        self.create(robot.root, Transformation())
+        self.create()
         self.scale_factor = 1.
 
-    def transform(self, native_mesh, transformation):
-        """Transforms a CAD-specific mesh using a **COMPAS** transformation.
+    def transform(self, geometry, transformation):
+        """Transforms a CAD-specific geometry using a **COMPAS** transformation.
 
         Parameters
         ----------
-        native_mesh : object
-            A CAD-specific (i.e. native) geometry object as returned by :meth:`draw_mesh`.
+        geometry : object
+            A CAD-specific (i.e. native) geometry object as returned by :meth:`draw_geometry`.
         transformation : `Transformation`
             **COMPAS** transformation to update the geometry object.
         """
         raise NotImplementedError
 
-    def draw_mesh(self, mesh, color=None):
-        """Draw a **COMPAS** mesh into the CAD environment.
+    def draw_geometry(self, geometry, color=None):
+        """Draw a **COMPAS** geometry in the respective CAD environment.
 
         Note
         ----
         This is an abstract method that needs to be implemented by derived classes.
 
         Parameters
         ----------
-        mesh : :class:`compas.datastructures.Mesh`
+        geometry : :class:`compas.datastructures.Mesh`
```

Comment:
If the name changed to `draw_geometry`, does that mean we should change the input data type from `compas.datastructures.Mesh` to something else? (although, there's no root class for compas geometry)

Reply:
Well, actually it should support primitives such as Box, Cylinder, Sphere, Capsule (yet to be implemented in compas.primitives) and Mesh. 
@brgcode: Here it would make sense to have a Base class for the geometry.primitives then we could use either `compas.datastructures.Mesh` or `compas.geometry.Primitive` as input data type.

Reply:
changed to `compas.datastructures.Mesh` or `compas.datastructures.Shape` in description